### PR TITLE
Beginnings of the AWS backend!

### PR DESCRIPTION
**Pull request by cjllanwarne**
Intended to avoid multiple people iterating on this having to regenerate the same set of placeholders:
 - Project definition in sbt
 - Basic ecs backend factory
 - Better defaults in the BackendLifecycleActorFactory trait
 - Fixed a few typos in nearby files... 😳

### CODE REVIEW

**Comment by Horneth** [comment]
👍

File: build.sbt [hunk 1]
```diff
@@ -43,6 +43,12 @@ lazy val sfsBackend = (project in backendRoot / "sfs")
   .dependsOn(gcsFileSystem)
   .dependsOn(backend % "test->test")
 
+lazy val ecsBackend = (project in backendRoot / "ecs")
```
Comment (mcovarr) [inline build.sbt]: Maybe `amazonBackend` would be more intuitive?  Unless we expect multiple Amazon-flavored backends.

**Comment by mcovarr** [comment]
:+1:

File: build.sbt (hunk 1 above)
Comment (kcibul) [inline build.sbt]: I think ecs is appropriate for now.  it's possible we might have other backends (and even this one might not be called ecs...)  But if we need to change it later, it wouldn't be the end of the world

Comment (kcibul) [inline build.sbt]: AWESOME to see this started though!!

Comment (mcovarr) [inline build.sbt]: Thirded, the motion carries.

**Pull request merged**